src/compiler/checker.ts

```diff
@@ -3924,6 +3924,12 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             else if (result.flags & SymbolFlags.RegularEnum) {
                 diagnosticMessage = error(errorLocation, Diagnostics.Enum_0_used_before_its_declaration, declarationName);
             }
+            else {
+                Debug.assert(!!(result.flags & SymbolFlags.ConstEnum));
+                if (compilerOptions.isolatedModules) {

```

Comment:
```suggestion
                if (getIsolatedModules(compilerOptions)) {
```

Reply:
I would commit this suggestion for you and approve, but GitHub can't handle this on checker.ts 🥲

Reply:
🫠

I pushed out the change + the relevant missing test case for `verbatimModuleSyntax` :p